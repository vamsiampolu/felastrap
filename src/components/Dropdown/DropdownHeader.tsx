import { useFela } from "react-fela";
import { Header, type HeaderProps } from "../Header/Header";

export const DropdownHeader = ({
  className = "",
  as = "h6",
  children,
  ...other
}: Partial<HeaderProps>) => {
  const { css } = useFela();
  const cx = `${css({
    color: "#6c757d",
    display: "block",
    fontSize: "0.875rem",
    marginBottom: 0,
    padding: "0.5rem 1rem",
    whiteSpace: "nowrap",
  })} dropdown-header`;

  return (
    <li data-testid="dropdown-header-item">
      <Header
        data-testid="dropdown-header"
        as={as}
        className={`${cx} ${className || ""}`}
        {...other}
      >
        {children}
      </Header>
    </li>
  );
};
