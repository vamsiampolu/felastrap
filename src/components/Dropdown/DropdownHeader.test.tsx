import { IRenderer } from "fela";
import { DropdownHeader } from "./DropdownHeader";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../test-helpers";
import { render, screen } from "@testing-library/react";

describe("components/Dropdown/DropdownHeader", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a Dropdown Header", async () => {
    const component = <DropdownHeader>Dropdown Header</DropdownHeader>;
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("dropdown-header-item")).toBeInTheDocument();
    expect(screen.getByTestId("dropdown-header")).toBeInTheDocument();

    expect(screen.queryByTestId("dropdown-header")).toHaveClass(
      "dropdown-header",
    );
    expect(screen.queryByTestId("dropdown-header")).toHaveTextContent(
      "Dropdown Header",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
