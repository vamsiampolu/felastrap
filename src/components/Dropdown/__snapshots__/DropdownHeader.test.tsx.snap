// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Dropdown/DropdownHeader should render a Dropdown Header 1`] = `
".a {
  color: #6c757d;
}
.b {
  display: block;
}
.c {
  font-size: 0.875rem;
}
.d {
  margin-bottom: 0;
}
.e {
  white-space: nowrap;
}
.f {
  padding-top: 0.5rem;
}
.g {
  padding-right: 1rem;
}
.h {
  padding-bottom: 0.5rem;
}
.i {
  padding-left: 1rem;
}
.j {
  color: inherit;
}
.k {
  font-weight: 500;
}
.l {
  font-size: 1rem;
}
.m {
  line-height: 1.2;
}
.n {
  margin-top: 0;
}
.o {
  margin-bottom: 0.5rem;
}
@media (min-width: 1200px) {
  .p {
    font-size: 1rem;
  }
}


<li data-testid="dropdown-header-item">
  <h6 class="a b c d e f g h i dropdown-header" data-testid="dropdown-header">
    Dropdown Header
  </h6>
</li>
;
"
`;
